web-runtime/npm-package/src/backgroundScript.ts

```diff
@@ -0,0 +1,39 @@
+import { expose } from 'threads';
+import { Envs, FaaSConfig } from './config';
+import { FaaS } from './FaaS';
+import { IFluenceAppService } from './types';
+
+const faasInstances = new Map<string, FaaS>();
+let controlModule: WebAssembly.Module;
+
+const toExpose: IFluenceAppService = {
+    init: async (marineWasm: SharedArrayBuffer): Promise<void> => {
+        controlModule = await WebAssembly.compile(new Uint8Array(marineWasm));
+    },
+    createService: async (
+        wasm: SharedArrayBuffer,
+        serviceId: string,
+        faaSConfig?: FaaSConfig,
+        envs?: Envs,
+    ): Promise<void> => {
+        const service = await WebAssembly.compile(new Uint8Array(wasm));
```

Comment:
suggest to call it just a module, and it'd actually a module on the 4th stage of marine-web.